src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3841,7 +3838,10 @@ void ride_construction_tooldown_construct(int32_t screenX, int32_t screenY)
         return;
     }
 
-    for (int32_t zAttempts = 41; zAttempts >= 0; zAttempts--)
+    // search for z value to build at, up to max ride height
+    int numAttempts = (z <= (8 * MAX_TRACK_HEIGHT) ? MAX_TRACK_HEIGHT - (z / 8) + 1 : 2);
```

Comment:
If you were to have MAX_TRACK_HEIGHT by default of multiple by 8 you can apply following:
```suggestion
    int numAttempts = std::clamp(MAX_TRACK_HEIGHT - z, 2, MAX_TRACK_HEIGHT );
```